src/components/Dashboard/TodoList.vue

```diff
@@ -1487,19 +1480,24 @@ export default class TodoList extends Vue {
       case FilingTypes.RESTORATION: {
         let restorationType: string
 
-        if (item.filingSubType === RestorationTypes.FULL) {
+        /**
+         * Type assertion is done to fix TypeScript error.
+         * "This condition will always return 'false' since the types
+         * 'FilingSubTypes' and 'RestorationTypes' have no overlap."
+         */
+        if ((item.filingSubType as any) === (RestorationTypes.FULL as any)) {
           restorationType = RestorationTypes.FULL
         }
 
-        if (item.filingSubType === RestorationTypes.LIMITED) {
+        if ((item.filingSubType as any) === (RestorationTypes.LIMITED as any)) {
```

Comment:
Instead of going type-less here, I'd prefer to see the enum changed to `FilingSubTypes.FULL_RESTORATION`, etc.

Reply:
Also below in `buildRestorationUrl()`.

Reply:
I'm reverting this change. It seems to me that there were a lot of stuff that Vue was complaining about but not anymore.

I'm not really sure if it's related to the Vue version or to the rules that I've added at a certain point. I'll be going through the code again and see what changes I can revert back since they might not be a problem anymore.

Reply:
If I change to `FilingSubTypes.FULL_RESTORATION`, I'm getting the same complaint as before:
"This comparison appears to be unintentional because the types 'FilingSubTypes' and 'RestorationTypes' have no overlap."

Reply:
Weird. It works for me.

![image](https://github.com/bcgov/business-filings-ui/assets/10002889/eae6637a-a21a-499f-90ea-257929e0f528)

`restorationType` should also be FilingSubTypes, not a string (and also please fix buildRestorationUrl()).

Reply:
Yes, it's related to the Vue version -- the later versions do more thorough type-checking.

Reply:
![Screenshot (71)](https://github.com/bcgov/business-filings-ui/assets/122301442/4f61d351-d829-4abb-a1a9-0e57d7150b55)


I could leave it as is since the project is building successfully. What do you think?

Reply:
I forgot to remove the `as` in my screenshot above. Try this:

![image](https://github.com/bcgov/business-filings-ui/assets/10002889/b7dc0747-c764-4e1c-a7df-927dca78ea7c)

Reply:
> `restorationType` should also be FilingSubTypes, not a string (and also please fix buildRestorationUrl()).

I'd prefer to use actual types than string :)

Reply:
Oh oh sorry. My mind was thinking of something else. `restorationType` being of type RestorationTypes. OK Sev, on it. Will do that!

Reply:
Fixed, thanks!